aten/src/ATen/xpu/detail/XPUHooks.cpp

```diff
@@ -0,0 +1,35 @@
+#include <ATen/xpu/XPUContext.h>
+#include <ATen/xpu/XPUDevice.h>
+#include <ATen/xpu/detail/XPUHooks.h>
+#include <c10/util/CallOnce.h>
+
+namespace at::xpu::detail {
+
+void XPUHooks::initXPU() const {
+  // TODO: the initialization of device allocator should be placed here.
+}
+
+bool XPUHooks::hasXPU() const {
+  return true;
+}
+
+std::string XPUHooks::showConfig() const {
+  return "XPU backend";
+}
+
+DeviceIndex XPUHooks::getGlobalIdxFromDevice(const at::Device& device) const {
```

Comment:
Should we use int as return type? The global index is not DeviceIndex defined by PyTorch, right?
Instead, it's defined by DLPack? We cannot guarantee it always aligns to PyTorch definition?

Reply:
@guangyey , as we discussed, we should return int and add some comments here, right?

Reply:
Agreed. change to return int.